packages/twenty-front/src/modules/object-record/record-table/record-table-header/components/RecordTableHeaderCheckboxColumn.tsx

```diff
@@ -13,22 +13,30 @@ const StyledContainer = styled.div`
   background-color: ${({ theme }) => theme.background.primary};
 `;
 
-const StyledColumnHeaderCell = styled.th`
+const StyledColumnHeaderCell = styled.th<{ setNumColumns: number }>`
   background-color: ${({ theme }) => theme.background.primary};
   border-bottom: 1px solid ${({ theme }) => theme.border.color.light};
   border-right: transparent;
-  max-width: 30px;
-  min-width: 30px;
-  width: 30px;
+  text-align: center;
+  width: ${({ setNumColumns }) => {
+    return setNumColumns <= 6 ? '22px' : '32px';
```

Comment:
Hey @harshit078 thanks! This doesn't seem like something we'd merge (too hacky). No luck with pure-css solutions? That's what we'd want ideally

Reply:
Pure-css solution would make the table laggy and not the best approach !